subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -995,36 +1022,14 @@ int do_write_op_tlv(struct lwm2m_message *msg)
 					continue;
 				}
 
-				msg->path.res_id = tlv2.id;
-				msg->path.level = 3U;
-				ret = do_write_op_tlv_item(msg);
-				/*
-				 * ignore errors for CREATE op
-				 * for OP_CREATE and BOOTSTRAP WRITE: errors on
-				 * optional resources are ignored (ENOTSUP)
-				 */
-				if (ret < 0 &&
-				    !((ret == -ENOTSUP) &&
-				      (msg->ctx->bootstrap_mode ||
-				       msg->operation == LWM2M_OP_CREATE))) {
+				if ((ret = write_tlv_resource(msg, &tlv2))) {
```

Comment:
We typically avoid assignments inside the `if` condition
```suggestion
				ret = write_tlv_resource(msg, &tlv2);
				if (ret < 0) {
```